tools/trace_analyzer.py

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python3
+
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright(c) 2023 Intel Corporation. All rights reserved.
+
+'''
+A tool to analyze SOF trace file.
+'''
+
+import re
+import argparse
+from collections import namedtuple
+
+CPU_CLK = 400000000
+CPU_TIMER = 38400000
+AUDIO_PERIOD = 1000
+TICK_TO_MCPS = CPU_CLK / CPU_TIMER / AUDIO_PERIOD
+
+TraceItem = namedtuple('TraceItem', ['timestamp', 'level', 'context', 'func', 'msg'])
+CompId = namedtuple('CompId', ['ppl_id', 'comp_id'])
+CompPerfInfo = namedtuple('CompPerfInfo', ['timestamp', 'samples', 'period', 'cpu_avg', 'cpu_peak'])
+CompPerfStats = namedtuple('CompPerfStats', ['cpu_avg', 'cpu_peak'])
+
+class TraceAnalyzer:
+    '''
+    The TraceAnalyzer class for SOF trace analysis, provides statics for below items:
+        - SOF component performance
+    '''
+    def __init__(self):
+        self._perf_info = {}
+        self._perf_stats = {}
+
+    def _add_perf_trace(self, trace_item: TraceItem):
+        # Example of msg: ['comp:0', '0x20000', 'perf', 'comp_copy', 'samples', '48',
+        # period', '1000', 'cpu', 'avg', '322', 'peak', '333']
```

Comment:
Can you give the _unparsed_ example instead? Otherwise it's impossible to review the parsing code.

Also, if the input format changes then the unparsed example can be compared to the failing input.